pkg/awsiamauth/awsiamauth.go

```diff
@@ -0,0 +1,78 @@
+package awsiamauth
```

Comment:
Actually I named it like that because awsiam is kinda different from IAM authenticator :)
I wanted to keep it specific to authenticator.
Also i thought i might use awsiam name for some e2e tests packages, which I have not gotten to yet.